pkg/controllers/work/applier_test.go

```diff
@@ -0,0 +1,420 @@
+				Type:                  placementv1beta1.ApplyStrategyTypeServerSideApply,
+				ServerSideApplyConfig: &placementv1beta1.ServerSideApplyConfig{ForceConflicts: false},
+			},
+			ownerRefs: []metav1.OwnerReference{
+				{
+					APIVersion: placementv1beta1.GroupVersion.String(),
+					Kind:       placementv1beta1.AppliedWorkKind,
+					Name:       "work1",
+				},
+				{
+					APIVersion: placementv1beta1.GroupVersion.String(),
+					Kind:       placementv1beta1.AppliedWorkKind,
+					Name:       "work2",
+				},
+			},
+			works: []placementv1beta1.Work{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: testWorkNamespace,
+						Name:      "work1",
+					},
+					Spec: placementv1beta1.WorkSpec{
+						ApplyStrategy: &placementv1beta1.ApplyStrategy{
+							Type: placementv1beta1.ApplyStrategyTypeServerSideApply,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: testWorkNamespace,
+						Name:      "work2",
+					},
+					Spec: placementv1beta1.WorkSpec{
+						ApplyStrategy: &placementv1beta1.ApplyStrategy{
+							Type:                  placementv1beta1.ApplyStrategyTypeServerSideApply,
+							ServerSideApplyConfig: &placementv1beta1.ServerSideApplyConfig{ForceConflicts: true},
+						},
+					},
+				},
+			},
+			wantWorkName: "work2",
+		},
+		{
+			name: "work not found",
+			applyStrategy: placementv1beta1.ApplyStrategy{
+				Type:                  placementv1beta1.ApplyStrategyTypeServerSideApply,
+				ServerSideApplyConfig: &placementv1beta1.ServerSideApplyConfig{ForceConflicts: false},
+			},
+			ownerRefs: []metav1.OwnerReference{
+				{
+					APIVersion: placementv1beta1.GroupVersion.String(),
+					Kind:       placementv1beta1.AppliedWorkKind,
+					Name:       "work1",
+				},
+			},
+			wantErr: controller.ErrExpectedBehavior,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			var objects []client.Object
+			for i := range tc.works {
+				objects = append(objects, &tc.works[i])
+			}
+			scheme := serviceScheme(t)
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				Build()
+			got, err := findConflictedWork(ctx, fakeClient, testWorkNamespace, &tc.applyStrategy, tc.ownerRefs)
+			if gotErr, wantErr := err != nil, tc.wantErr != nil; gotErr != wantErr || !errors.Is(err, tc.wantErr) {
+				t.Fatalf("findConflictedWork() got error %v, want error %v", err, tc.wantErr)
+			}
+			if tc.wantErr != nil {
+				return
+			}
+			if got == nil && tc.wantWorkName != "" || got != nil && got.Name != tc.wantWorkName {
+				t.Errorf("findConflictedWork() got %v, want %v", got.Name, tc.wantWorkName)
+			}
+		})
+	}
+}
+
+func TestValidateOwnerReference(t *testing.T) {
+	tests := []struct {
+		name          string
+		applyStrategy placementv1beta1.ApplyStrategy
+		ownerRefs     []metav1.OwnerReference
+		works         []placementv1beta1.Work
+		want          ApplyAction
```

Comment:
Just a nit: maybe `wantAction` is better, considering that there's already a `wantErr`?

Reply:
I usually name the output (excluding error) when there are multiple ones. I feel it's ok for the single output.